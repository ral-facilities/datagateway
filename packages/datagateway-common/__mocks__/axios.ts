import axios from 'axios';

const requests = {
  get: vi.fn(() => Promise.resolve({ data: {} })),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  delete: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  AxiosError: axios.AxiosError,
  isAxiosError: axios.isAxiosError,
};

export default requests;
