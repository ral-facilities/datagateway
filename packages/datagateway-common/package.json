{
  "name": "datagateway-common",
  "version": "1.1.1",
  "private": true,
  "files": [
    "lib"
  ],
  "main": "./lib/index.js",
  "dependencies": {
    "@date-io/date-fns": "2.15.0",
    "@emotion/react": "11.10.0",
    "@emotion/styled": "11.10.0",
    "@mui/x-date-pickers": "5.0.0-beta.5",
    "@types/lodash.debounce": "4.0.6",
    "axios": "0.27.2",
    "connected-react-router": "6.9.1",
    "date-fns": "2.29.1",
    "hex-to-rgba": "2.0.1",
    "history": "4.10.1",
    "i18next": "21.9.1",
    "lodash.debounce": "4.0.8",
    "loglevel": "1.8.0",
    "prettier": "2.7.1",
    "react-draggable": "4.4.3",
    "react-i18next": "11.18.3",
    "react-query": "3.39.2",
    "react-redux": "7.2.6",
    "react-scripts": "5.0.0",
    "react-virtualized": "9.22.3",
    "redux": "4.2.0",
    "redux-mock-store": "1.5.4",
    "redux-thunk": "2.4.1",
    "resize-observer-polyfill": "1.5.1",
    "tslib": "2.4.0",
    "typescript": "4.7.4",
    "use-deep-compare-effect": "1.8.1"
  },
  "peerDependencies": {
    "@mui/icons-material": ">= 5.5.0 < 6",
    "@mui/material": ">= 5.5.0 < 6",
    "react": ">= 17.0.2 < 18",
    "react-dom": ">= 17.0.2 < 18",
    "react-router-dom": ">= 5.2.0 < 6"
  },
  "devDependencies": {
    "@mui/icons-material": "5.10.3",
    "@mui/material": "5.10.0",
    "@testing-library/jest-dom": "5.16.4",
    "@testing-library/react": "12.1.3",
    "@testing-library/react-hooks": "8.0.1",
    "@testing-library/user-event": "14.4.1",
    "@types/enzyme": "3.10.10",
    "@types/jest": "27.5.2",
    "@types/node": "17.0.17",
    "@types/react": "17.0.39",
    "@types/react-router-dom": "5.3.3",
    "@types/react-virtualized": "9.21.10",
    "@typescript-eslint/eslint-plugin": "5.32.0",
    "@typescript-eslint/parser": "5.32.0",
    "@wojtekmaj/enzyme-adapter-react-17": "0.6.6",
    "babel-eslint": "10.1.0",
    "enzyme": "3.11.0",
    "enzyme-to-json": "3.6.1",
    "eslint": "8.23.0",
    "eslint-config-prettier": "8.5.0",
    "eslint-config-react-app": "7.0.0",
    "eslint-plugin-cypress": "2.12.1",
    "eslint-plugin-prettier": "4.2.1",
    "lint-staged": "13.0.3",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-router-dom": "5.3.0",
    "react-test-renderer": "17.0.2"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom --coverage --watchAll=false",
    "postinstall": "yarn tsc",
    "test:watch": "react-scripts test --env=jsdom --watch",
    "lint:js": "eslint --ext=tsx --ext=ts --ext=js --ext=jsx --fix ./src",
    "tsc": "tsc -p tsconfig.lib.json",
    "eject": "react-scripts eject",
    "pre-commit": "lint-staged"
  },
  "lint-staged": {
    "src/**/*.{tsx,ts,js,jsx,json}": [
      "eslint --max-warnings=0 --ext=tsx --ext=ts --ext=js --ext=jsx  --fix",
      "prettier --write"
    ],
    "cypress/**/*.{tsx,ts,js,jsx,json}": [
      "eslint --max-warnings=0 --ext=tsx --ext=ts --ext=js --ext=jsx --fix",
      "prettier --write"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ]
  },
  "jest": {
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "collectCoverageFrom": [
      "src/**/*.{tsx,ts,js,jsx}",
      "!src/index.tsx",
      "!src/serviceWorker.ts",
      "!src/setupTests.tsx"
    ],
    "resetMocks": false
  }
}
