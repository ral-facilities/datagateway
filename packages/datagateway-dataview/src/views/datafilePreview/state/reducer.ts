import DATAFILE_PREVIEWER_DEFAULT from '../defaults';
import {
  DecrementDatafilePreviewerZoomLevelType,
  IncrementDatafilePreviewerZoomLevelType,
  ResetDatafilePreviewerZoomLevelType,
  ToggleDatafilePreviewerDetailsPanePayload,
  ToggleDatafilePreviewerDetailsPaneType,
} from './actions';
import { DGDataViewState } from '../../../state/app.types';

/**
 * Defines the shape of the redux state for {@link DatafilePreviewer}
 */
interface DatafilePreviewerState {
  /**
   * Whether the details pane should be shown or not.
   */
  isDetailsPaneShown: boolean;

  /**
   * The current zoom level of the previewer, in percentage. For example,
   * 100% maps to 100, 110% maps to 110, 80% maps to 80, etc.
   */
  zoomLevel: number;
}

function toggleDatafilePreviewerDetailsPane(
  state: DGDataViewState,
  payload: ToggleDatafilePreviewerDetailsPanePayload
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      isDetailsPaneShown: payload.shouldShow,
    },
  };
}

function incDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      zoomLevel:
        state.datafilePreviewer.zoomLevel +
        DATAFILE_PREVIEWER_DEFAULT.zoomLevelStep,
    },
  };
}

function decDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      // zoom level should not go lower than zoomLevelStep
      // otherwise will result in zoom level <= 0.
      zoomLevel: Math.max(
        DATAFILE_PREVIEWER_DEFAULT.minZoomLevel,
        state.datafilePreviewer.zoomLevel -
          DATAFILE_PREVIEWER_DEFAULT.zoomLevelStep
      ),
    },
  };
}

function resetDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      zoomLevel: DATAFILE_PREVIEWER_DEFAULT.zoomLevel,
    },
  };
}

const datafilePreviewerInitialState: DatafilePreviewerState = {
  isDetailsPaneShown: DATAFILE_PREVIEWER_DEFAULT.showDetailsPane,
  zoomLevel: DATAFILE_PREVIEWER_DEFAULT.zoomLevel,
};

const datafilePreviewerReducer = {
  [ToggleDatafilePreviewerDetailsPaneType]: toggleDatafilePreviewerDetailsPane,
  [IncrementDatafilePreviewerZoomLevelType]: incDatafilePreviewerZoomLevel,
  [DecrementDatafilePreviewerZoomLevelType]: decDatafilePreviewerZoomLevel,
  [ResetDatafilePreviewerZoomLevelType]: resetDatafilePreviewerZoomLevel,
};

export {
  datafilePreviewerInitialState,
  datafilePreviewerReducer,
  toggleDatafilePreviewerDetailsPane,
  incDatafilePreviewerZoomLevel,
  decDatafilePreviewerZoomLevel,
  resetDatafilePreviewerZoomLevel,
};
export type { DatafilePreviewerState };
