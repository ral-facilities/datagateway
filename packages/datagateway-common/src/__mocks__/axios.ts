// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn(() => Promise.resolve({ data: {} })),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  AxiosError: axios.AxiosError,
};

export default requests;
