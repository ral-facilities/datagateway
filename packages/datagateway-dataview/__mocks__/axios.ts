import axios from 'axios';

const requests = {
  get: vi.fn((path) => {
    if (path === '/datagateway-dataview-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'Generic',
          idsUrl: 'ids',
          apiUrl: 'api',
          downloadApiUrl: 'download',
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  delete: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  isAxiosError: axios.isAxiosError,
};

export default requests;
