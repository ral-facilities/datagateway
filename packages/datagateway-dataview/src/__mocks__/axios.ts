// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn((path) => {
    if (path === '/datagateway-dataview-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'Generic',
          idsUrl: 'ids',
          apiUrl: 'api',
          downloadApiUrl: 'download',
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  isAxiosError: axios.isAxiosError,
};

export default requests;
