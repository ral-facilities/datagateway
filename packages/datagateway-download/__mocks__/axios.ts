import axios from 'axios';

const requests = {
  get: vi.fn((path) => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'https://example.com/api',
          downloadApiUrl: 'https://example.com/downloadApi',
          idsUrl: 'https://example.com/ids',
          fileCountMax: 5000,
          totalSizeMax: 1000000000000,
          accessMethods: {
            https: {
              idsUrl: 'https://example.com/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://example.com/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
          routes: [
            {
              section: 'Test',
              link: '/download',
              displayName: 'Download',
              order: 0,
            },
          ],
        },
      });
    } else if (path.includes('/topcat/user/downloads')) {
      return Promise.resolve({ data: [] });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  put: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};

export default requests;
