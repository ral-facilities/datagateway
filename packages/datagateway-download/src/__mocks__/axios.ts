// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn((path) => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'https://example.com/api',
          downloadApiUrl: 'https://example.com/downloadApi',
          idsUrl: 'https://example.com/ids',
          fileCountMax: 5000,
          totalSizeMax: 1000000000000,
          accessMethods: {
            https: {
              idsUrl: 'https://example.com/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://example.com/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
          routes: [
            {
              section: 'Test',
              link: '/download',
              displayName: 'Download',
              order: 0,
            },
          ],
        },
      });
    } else if (path.includes('/topcat/user/downloads')) {
      return Promise.resolve({ data: [] });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  put: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};

export default requests;
