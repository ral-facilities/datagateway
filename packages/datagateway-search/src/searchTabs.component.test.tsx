import * as React from 'react';
import { StateType } from './state/app.types';
import { Provider } from 'react-redux';
import { mount, ReactWrapper } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter, Router } from 'react-router-dom';
import { dGCommonInitialState, CartProps } from 'datagateway-common';
import axios from 'axios';
import { Store } from 'redux';
import { createMemoryHistory, History } from 'history';
import { render } from '@testing-library/react';
import { QueryClientProvider, QueryClient } from 'react-query';

import SearchPageTabs, { SearchTabsProps } from './searchTabs.component';
import { initialState } from './state/reducers/dgsearch.reducer';
import InvestigationSearchTable from './table/investigationSearchTable/investigationSearchTable.component';
import DatasetSearchTable from './table/datasetSearchTable.component';
import DatafileSearchTable from './table/datafileSearchTable.component';
import InvestigationCardView from './card/investigationSearchCardView.component';
import DatasetCardView from './card/datasetSearchCardView.component';

jest.mock('datagateway-common', () => ({
  ...jest.requireActual('datagateway-common'),
  __esModule: true,
  // mock table to opt out of rendering them in these tests as there's no need
  Table: jest.fn(() => 'MockedTable'),
}));

describe('SearchTabs', () => {
  let state: StateType;
  let history: History;
  let props: SearchTabsProps & CartProps;
  const mockStore = configureStore([thunk]);

  const onTabChange = jest.fn();

  const createWrapper = (): ReactWrapper => {
    const store: Store = mockStore(state);
    return mount(
      <Provider store={store}>
        <Router history={history}>
          <QueryClientProvider client={new QueryClient()}>
            <SearchPageTabs {...props} />
          </QueryClientProvider>
        </Router>
      </Provider>
    );
  };

  beforeEach(() => {
    history = createMemoryHistory({
      initialEntries: ['/search/data'],
    });

    state = JSON.parse(
      JSON.stringify({ dgsearch: initialState, dgcommon: dGCommonInitialState })
    );

    props = {
      view: 'table',
      onTabChange: onTabChange,
      currentTab: 'investigation',
      cartItems: [],
      navigateToDownload: jest.fn(),
    };

    (axios.get as jest.Mock).mockImplementation((url) => {
      if (url.includes('count')) {
        return Promise.resolve({ data: 0 });
      } else {
        return Promise.resolve({ data: [] });
      }
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when request received', async () => {
    state.dgsearch = {
      ...state.dgsearch,
      tabs: {
        datasetTab: true,
        datafileTab: true,
        investigationTab: true,
      },
    };
    (axios.get as jest.Mock).mockImplementation((url) => {
      if (url.includes('count')) {
        return Promise.resolve({ data: 1 });
      } else {
        console.log('sdakjkjsakldjaksdklad');
        return Promise.resolve({ data: Array(1) });
      }
    });

    const testStore = mockStore(state);
    const { asFragment } = render(
      <Provider store={testStore}>
        <MemoryRouter
          initialEntries={[{ key: 'testKey', pathname: '/search/data' }]}
        >
          <QueryClientProvider client={new QueryClient()}>
            <SearchPageTabs {...props} />
          </QueryClientProvider>
        </MemoryRouter>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('changes selected tab value on click of a new tab', () => {
    state.dgsearch = {
      ...state.dgsearch,
      tabs: {
        datasetTab: true,
        datafileTab: true,
        investigationTab: true,
      },
    };

    const wrapper = createWrapper();

    expect(wrapper.exists(InvestigationSearchTable)).toBeTruthy();

    wrapper
      .find('[aria-controls="simple-tabpanel-dataset"]')
      .last()
      .simulate('click');

    expect(onTabChange).toHaveBeenNthCalledWith(1, 'dataset');
  });

  describe('table view', () => {
    it('has the investigation search table component when on the investigation tab', () => {
      const wrapper = createWrapper();
      expect(wrapper.exists(InvestigationSearchTable)).toBeTruthy();
    });

    it('has the dataset search table component when on the dataset tab', () => {
      props.currentTab = 'dataset';

      // Mock to prevent error logging
      jest.spyOn(console, 'error').mockImplementation();
      const wrapper = createWrapper();

      expect(wrapper.exists(DatasetSearchTable)).toBeTruthy();
    });

    it('has the datafile search table component when on the datafile tab', () => {
      props.currentTab = 'datafile';

      // Mock to prevent error logging
      jest.spyOn(console, 'error').mockImplementation();
      const wrapper = createWrapper();

      expect(wrapper.exists(DatafileSearchTable)).toBeTruthy();
    });
  });

  describe('card view', () => {
    beforeEach(() => {
      props.view = 'card';
    });

    it('has the investigation search card view component when on the investigation tab', () => {
      const wrapper = createWrapper();
      expect(wrapper.exists(InvestigationCardView)).toBeTruthy();
    });

    it('has the dataset search card view component when on the dataset tab', () => {
      props.currentTab = 'dataset';

      // Mock to prevent error logging
      jest.spyOn(console, 'error').mockImplementation();
      const wrapper = createWrapper();

      expect(wrapper.exists(DatasetCardView)).toBeTruthy();
    });

    it('has the datafile search table component when on the datafile tab', () => {
      props.currentTab = 'datafile';

      // Mock to prevent error logging
      jest.spyOn(console, 'error').mockImplementation();
      const wrapper = createWrapper();

      expect(wrapper.exists(DatafileSearchTable)).toBeTruthy();
    });
  });
});
