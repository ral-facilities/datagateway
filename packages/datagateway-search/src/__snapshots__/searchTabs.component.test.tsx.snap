// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTabs renders correctly when request received 1`] = `
<DocumentFragment>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-120r8wl-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiBox-root css-12nzfz8"
      >
        <div
          class="MuiTabs-root tour-search-tab-select css-1sl6kf8-MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-bottom: 0px;"
          >
            <div
              aria-label="searchPageTable.tabs_arialabel"
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-controls="simple-tabpanel-investigation"
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
                id="simple-tab-investigation"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiBadge-root BaseBadge-root css-b49g95-MuiBadge-root"
                  id="investigation-badge"
                >
                  <span
                    style="padding-right: 1ch; font-size: 16px; font-weight: bold;"
                  >
                    tabs.investigation
                  </span>
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular BaseBadge-badge css-fvc8ir-MuiBadge-badge"
                  >
                    <span
                      style="font-size: 14px; font-weight: bold; margin-top: 1px;"
                    >
                      0
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="simple-tabpanel-dataset"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
                id="simple-tab-dataset"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiBadge-root BaseBadge-root css-b49g95-MuiBadge-root"
                  id="dataset-badge"
                >
                  <span
                    style="padding-right: 1ch; font-size: 16px; font-weight: bold;"
                  >
                    tabs.dataset
                  </span>
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular BaseBadge-badge css-fvc8ir-MuiBadge-badge"
                  >
                    0
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-controls="simple-tabpanel-datafile"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
                id="simple-tab-datafile"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiBadge-root BaseBadge-root css-b49g95-MuiBadge-root"
                  id="datafile-badge"
                >
                  <span
                    style="padding-right: 1ch; font-size: 16px; font-weight: bold;"
                  >
                    tabs.datafile
                  </span>
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular BaseBadge-badge css-fvc8ir-MuiBadge-badge"
                  >
                    0
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-11yukd5-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-shc6j8"
        >
          <button
            aria-label="app.cart_arialabel"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium tour-dataview-cart-icon css-4e50t7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary BaseBadge-badge css-106c1u2-MuiBadge-badge"
              />
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <div
      aria-labelledby="simple-tab-investigation"
      class="MuiBox-root css-1k9h2vf"
      id="simple-tabpanel-investigation"
      role="tabpanel"
    >
      <div
        class="MuiBox-root css-zoser8"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1dgr0mp-MuiPaper-root"
        >
          MockedTable
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
